graylog2-web-interface/src/views/logic/views/UseCreateViewForEventDefinition.test.ts

```diff
@@ -0,0 +1,125 @@
+import generateId from 'logic/generateId';
+import asMock from 'helpers/mocking/AsMock';
+import type View from 'views/logic/views/View';
+
+const counter = () => {
+  let index = 0;
+
+  return () => {
+    // eslint-disable-next-line no-plusplus
+    return index++;
+  };
+};
+
+const generateIdCounterTwoAggregations = counter();
+const objectIdCounterTwoAggregations = counter();
+const generateIdCounterOneAggregation = counter();
+const objectIdCounterOneAggregations = counter();
+
+const mockedGenerateIdTwoAggregations = () => {
+  const idSet = ['query-id', 'mc-widget-id', 'allm-widget-id', 'field1-widget-id', 'field2-widget-id', 'summary-widget-id'];
+  const index = generateIdCounterTwoAggregations();
+
+  return idSet[index];
+};
+
+const mockedObjectIdTwoAggregations = () => {
+  const idSet = ['', 'view-id', 'search-id'];
+  const index = objectIdCounterTwoAggregations();
+
+  return idSet[index];
+};
+
+const mockedObjectIdOneAggregation = () => {
+  const idSet = ['', 'view-id', 'search-id'];
+  const index = objectIdCounterOneAggregations();
+
+  return idSet[index];
+};
+
+const mockedGenerateIdOneAggregation = () => {
+  const idSet = ['query-id', 'mc-widget-id', 'allm-widget-id', 'field1-widget-id'];
+  const index = generateIdCounterOneAggregation();
+
+  return idSet[index];
+};
+
+jest.mock('graylog-web-plugin/plugin', () => ({
+  PluginStore: { exports: jest.fn(() => [{ type: 'aggregation', defaults: {} }]) },
+}));
+
+jest.mock('logic/generateId', () => jest.fn());
+
+jest.mock('bson-objectid', () => jest.fn());
+
+jest.mock('views/logic/Widgets', () => ({
+  ...jest.requireActual('views/logic/Widgets'),
+  widgetDefinition: () => ({
+    searchTypes: () => [{
+      type: 'AGGREGATION',
+      typeDefinition: {},
+    }],
+  }),
+}));
+
+const todayDate = new Date();
+const withCurrentDate = (view: View) => view.toBuilder().createdAt(todayDate).build();
+
+describe('UseCreateViewForEventDefinition', () => {
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should create view with 2 aggregation widgets and one summary', async () => {
+    asMock(generateId).mockImplementation(mockedGenerateIdTwoAggregations);
+
+    // @ts-ignore
+    asMock(ObjectID).mockImplementation(() => ({
+      toString: () => mockedObjectIdTwoAggregations(),
+    }));
+
+    const { result } = renderHook(() => UseCreateViewForEventDefinition({ eventDefinition: mockEventDefinitionTwoAggregations, aggregations: mockedMappedAggregation }));
+    const view = await result.current.then((r) => r);
+
+    expect(withCurrentDate(view)).toEqual(withCurrentDate(mockedViewWithTwoAggregationsED));
+  });
+
+  it('should create view with 1 aggregation widgets and without summary', async () => {
+    asMock(generateId).mockImplementation(mockedGenerateIdOneAggregation);
+
+    // @ts-ignore
+    asMock(ObjectID).mockImplementation(() => ({
+      toString: () => mockedObjectIdOneAggregation(),
+    }));
```

Comment:
Instead of using ` // @ts-ignore` which ignores all possible errors, you can do:
```
asMock(ObjectID).mockImplementation(() => ({
      toString: () => mockedObjectIdOneAggregation(),
} as ObjectID));
```